website/docs/guides/markdown-features/markdown-features-admonitions.mdx

```diff
@@ -248,9 +249,70 @@ module.exports = {
 };
 ```
 
-The plugin accepts two options:
+The plugin accepts the following options:
 
 - `tag`: The tag that encloses the admonition. Defaults to `:::`.
-- `keywords`: An array of keywords that can be used as the type for the admonition. Note that if you override this, the default values will not be applied.
+- `keywords`: An array of keywords that can be used as the type for the admonition.
+- `appendKeywords`: Should the provided option `keywords` be merged into the existing default keywords. Defaults to `true`.
 
-The `keyword` will be passed as the `type` prop of the `Admonition` component. If you register more types than the default, you are also responsible for providing their implementation—including the container's style, icon, default title text, etc. You would usually need to [eject](../../swizzling.md#ejecting) the `@theme/Admonition` component, so you could re-use the same infrastructure as the other types.
+The `keyword` will be passed as the `type` prop of the `Admonition` component.
+
+### Custom admonition type components {#custom-admonition-type-components}
+
+By default, the theme doesn't know what do to with custom admonition keywords such as `:::my-custom-admonition`. It is your responsibility to map each admonition keyword to a React component so that the theme knows how to render them.
+
+Let's consider the following config:
+
+```js
+const admonitionOptions = {
+  tag: ':::',
+  keywords: ['my-custom-admonition'],
+  appendKeywords: true,
+};
+```
+
+You can register a custom React component for `:::my-custom-admonition` by creating the following file:
```

Comment:
```suggestion
You can provide the corresponding React component for `:::my-custom-admonition` by creating the following file (unfortunately, since it's not a React component file, it's not swizzlable):
```